cmd/asset-syncer/utils.go

```diff
@@ -318,57 +329,89 @@ func (o *ociAPICli) TagList(appName string) (*TagList, error) {
 	return &appTags, nil
 }
 
-// FilterChartTags modifies tagList and remove artifacts in other formats (e.g. Docker images)
-func (o *ociAPICli) FilterChartTags(tagList *TagList) error {
-	url := *o.url
-	chartTags := []string{}
-	for _, tag := range tagList.Tags {
-		url.Path = path.Join("v2", tagList.Name, "manifests", tag)
-		log.Debugf("getting tag %s", url.String())
-		manifestData, err := doReq(
-			url.String(),
-			map[string]string{
-				"Authorization": o.authHeader,
-				"Accept":        "application/vnd.oci.image.manifest.v1+json",
-			})
-		if err != nil {
-			return err
-		}
-		var manifest OCIManifest
-		err = json.Unmarshal(manifestData, &manifest)
-		if err != nil {
-			return err
-		}
-		if manifest.Config.MediaType == helm.HelmChartConfigMediaType {
-			chartTags = append(chartTags, tag)
-		}
+func (o *ociAPICli) IsHelmChart(appName, tag string) (bool, error) {
+	repoURL := *o.url
+	repoURL.Path = path.Join("v2", repoURL.Path, appName, "manifests", tag)
+	log.Debugf("getting tag %s", repoURL.String())
+	manifestData, err := doReq(
+		repoURL.String(),
+		map[string]string{
+			"Authorization": o.authHeader,
+			"Accept":        "application/vnd.oci.image.manifest.v1+json",
+		})
+	if err != nil {
+		return false, err
+	}
+	var manifest OCIManifest
+	err = json.Unmarshal(manifestData, &manifest)
+	if err != nil {
+		return false, err
+	}
+	return manifest.Config.MediaType == helm.HelmChartConfigMediaType, nil
+}
+
+func tagCheckerWorker(o ociAPI, tagJobs <-chan checkTagJob, resultChan chan checkTagResult) {
+	for j := range tagJobs {
+		isHelmChart, err := o.IsHelmChart(j.AppName, j.Tag)
+		resultChan <- checkTagResult{j, isHelmChart, err}
 	}
-	tagList.Tags = chartTags
-	return nil
 }
 
 // Checksum returns the sha256 of the repo by concatenating tags for
 // all repositories within the registry and returning the sha256.
 // Caveat: Mutated image tags won't be detected as new
 func (r *OCIRegistry) Checksum() (string, error) {
-	tagList := map[string]TagList{}
+	r.tags = map[string]TagList{}
+	checktagJobs := make(chan checkTagJob, numWorkers)
+	tagcheckRes := make(chan checkTagResult)
+	var wg sync.WaitGroup
+	wg.Add(1)
```

Comment:
So similar comment here - this isn't how a wait group should be used. See what you think. The result should be the same - great work on getting it down to 4m!